src/fontra/views/editor/editor.js

```diff
@@ -421,6 +421,17 @@ export class EditorController {
       this.updateWindowLocationAndSelectionInfo();
       this.autoViewBox = false;
     });
+    this.sceneController.addEventListener("createNewSource", (event) => {
+      this.sidebarDesignspace.addSource();
+    });
+    this.sceneController.addEventListener("goToNearestSource", async (event) => {
+      const glyphController =
+        await this.sceneController.sceneModel.getSelectedVariableGlyphController();
+      const nearestSource = glyphController.findNearestSourceFromGlobalLocation(
+        this.sceneController.getLocation()
+      );
+      this.designspaceLocationController.model.location = nearestSource;
```

Comment:
`nearestSource` is a source index, not a location. So you'll need to find out the location for that source. I think this causes both issues I mentioned.

Reply:
Or actually, let me have a look, I think we need to just "select te source" by index. Perhaps we need to add a method to the sidebar code.

(I'm in the process of figuring out how to refactor some if this, especially the messy relationship between scene-controller and scene-model.)

Reply:
Thanks.